MovieViewer/FirebaseUserDataService.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  Firebase.swift
+//  MovieViewer
+//
+//  Created by hvmark on 7/11/16.
+//  Copyright © 2016 Dang Quoc Huy. All rights reserved.
+//
+
+import Firebase
+
+//class FirebaseUserDataService: UserDataServiceProtocol {
+class FirebaseUserDataService {
+
+    static func userFavoriteMovie(username: String, movieID: Int) {
+        getUserFavorites("hieuvo") { (userFavorites) in
+            guard !userFavorites.contains(movieID) else { return }
+            
+            var newFavorites = userFavorites
+            newFavorites.append(movieID)
+            
+            let firebaseRef = FIRDatabase.database().reference()
+            firebaseRef.child("users").child(username).updateChildValues(["favorites": newFavorites])
```

Comment:
If I understand correctly, your data looks like:

```
- users 
   - hieuvo: [1, 2, 3, 4, 5]
```

This is fine, but I think structuring it like:



```
- users 
   - hieuvo
      - 1: true
```
could be faster and more "firebase-y".